### PR TITLE
kernel: fix 9p issues

### PR DESCRIPTION
**Pull request by volth**
###### Motivation for this change

Fixes for 9P issues #23957 and #23020 until applied in upstream
The patches have been submitted to upstream by @dezgeg, one a month ago (still no reaction), another today.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by globin** [comment]
What about 4.10?

File: pkgs/os-specific/linux/kernel/9p-hack.patch [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ 
+diff --git a/net/9p/client.c b/net/9p/client.c
+--- a/net/9p/client.c
++++ b/net/9p/client.c
+@@ -749,7 +749,7 @@ p9_client_rpc(struct p9_client *c, int8_t type, const char *fmt, ...)
+ 	}
+ again:
+ 	/* Wait for the response */
+-	err = wait_event_interruptible(*req->wq,
++	err = wait_event_killable(*req->wq,
```
Comment (dezgeg) [inline pkgs/os-specific/linux/kernel/9p-hack.patch]: There are a bit more `wait_event_interruptible` in the virtio transport code. I will try the same replacement there as well and update the patches.

**Comment by dezgeg** [comment]
Applied in https://github.com/NixOS/nixpkgs/commit/ed41d50e9fe3d942cfde37e84de781c096309e5b.

**Comment by volth** [comment]
should it be backported ```release-17.03``` branch as well?

**Comment by dezgeg** [comment]
Yes, I will do that once I see there aren't any problems in Hydra.

**Comment by dezgeg** [comment]
Applied to stable in 6c5128c9c801645faa95db2dd3b36df24da39d33.

**Pull request closed without merging**